CMake/ResolveDependency.cmake

```diff
@@ -40,6 +40,8 @@ set(CMAKE_TLS_VERIFY true)
 
 macro(build_dependency dependency_name)
   string(TOLOWER ${dependency_name} dependency_name_lower)
+  set(CMAKE_COMPILER_SELECTION_ARGS -DCMAKE_C_COMPILER=${CMAKE_C_COMPILER}
```

Comment:
@czentgr can you try to `set(CMAKE_C_COMPILER ${CMAKE_C_COMPILER})` and `set(CMAKE_CXX_COMPILER ${CMAKE_CXX_COMPILER})` in the main CMakeLists.txt? That should make them global and pass to bundling.

Reply:
CMAKE_C_COMPILER/CMAKE_CXX_COMPILER is already set based on the CC/CXX environment variables. So we don;t need to override it with itself.
But, if they are set and picked up from the env that means the bundle project should also pick up CC/CXX and correspondingly set CMAKE_C_COMPILER/CMAKE_CXX_COMPILER for themselves.
I will double check this as we are relying on the user to set CC/CXX to use Clang15.